File: 2022/04/part1.rb
#!/usr/bin/env ruby

input = ARGV[0] == "ex" ? "example" : "input"

result = IO.foreach(File.join(__dir__, input), chomp: true).count do |line|
  a,b,c,d=line.split(/[-,]/).map(&:to_i)
  ((a..b).cover?(c..d) || (c..d).cover?(a..b))
end

p result

File: 2024/07/part1.py
#!/usr/bin/env python3

from pprint import pp
from sys import argv

def test(result: int, op: list[int]) -> bool:
	if len(op) == 1:
		return result == op[0]
	a, b, *rest = op
	return test(result, [a * b, *rest]) or test(result, [a + b, *rest])


with open(argv[1], "r") as f:
	count=0
	for line in f:
		a, b = line.strip().split(': ')
		a = int(a)
		b = [int(x) for x in b.split(' ')]
		# a, *b = [int(x) for x in re.split(r'[^\d]+', line.strip())]
		print(repr(a),repr(b))
		if test(a,b):
			count += a
	print(count)
